tests/step-103.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+# Test:
+#   Fail on not available shared hooks.
+
+if ! sh /var/lib/githooks/install.sh; then
+    echo "! Failed to execute the install script"
+    exit 1
+fi
+
+mkdir -p /tmp/shared/hooks-102.git/pre-commit &&
+    echo 'exit 0' >/tmp/shared/hooks-102.git/pre-commit/succeed &&
+    cd /tmp/shared/hooks-102.git &&
+    git init &&
+    git add . &&
+    git commit -m 'Initial commit' ||
+    exit 1
+
+# Install shared hook url into a repo.
+mkdir -p /tmp/test102 && cd /tmp/test102 || exit 1
+git init || exit 1
+mkdir -p .githooks && echo '/tmp/shared/hooks-102.git' >.githooks/.shared || exit 1
+git add .githooks/.shared
+git hooks shared update
+
+RESULT=$(ls -A ~/.githooks/shared 2>/dev/null)
+if [ "$RESULT" = "" ]; then
+    echo "! Expected shared hooks to be installed."
+    exit 1
+fi
+git commit -m "Test" || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
+
+# Clone a new one
+echo "Cloning"
+cd /tmp || exit 1
+git clone /tmp/test102 test102-clone && cd test102-clone || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
```

Comment:
This fails because you already purged, so there's nothing new to purge. It's like `rm something; rm something` then the second invocation is non-zero exit code.

Reply:
Then you remove line 41. The test on master should fail because Line 38(clone) does not install the shared hooks automatically, and any next commits pass through without running the shared hooks. This is really not wanted, we should really make if fail!
Thats mainly the problem. So removing Line 41. Should make it fail after Line 47.

Reply:
OK, so as far as I know, ~~there are no hooks run after a `git clone`~~. Oh, I see now that [post-checkout](https://git-scm.com/docs/githooks#_post_checkout) is run after `clone` normally as you found.

Currently, shared hooks are only refreshed at `post-merge` (i.e. when you `git pull` for example), or if you trigger explicitly through the cli tool. I had it like this originally, so that the shared repos are not updated over and over again slowly, when you're trying to do something quickly. Moving between branches with `git checkout` can be expected to complete quickly for example, so I wouldn't do a `git pull` on all the shared hook repos at that point. However, the docs say:
> It is also run after git-clone[1], unless the --no-checkout (-n) option is used. The first parameter given to the hook is the null-ref, the second the ref of the new HEAD and the flag is always 1. Likewise for git worktree add unless --no-checkout is used.

So if we can tell the `git clone` `post-checkout` events apart from the `git checkout` triggered ones, we could refresh shared hooks there as well.